tests/output.rs

```diff
@@ -1193,438 +980,54 @@ fn test_specify_explicit_file_of_unrecognized_file_type_and_language_flag() {
 }
 
 #[test]
-fn test_swift() {
-    assert_sorted_output(
-        "swift_project",
-        r#"
-            $ tree-sitter-grep -q '(value_argument) @c' --language swift
-            example.swift:2:    atPath: "native"
-        "#,
-    );
-}
-
-#[test]
-fn test_swift_auto_language() {
-    assert_sorted_output(
-        "swift_project",
-        r#"
-            $ tree-sitter-grep -q '(value_argument) @c'
-            example.swift:2:    atPath: "native"
-        "#,
-    );
-}
-
-#[test]
-fn test_objective_c() {
-    assert_sorted_output(
-        "objective_c_project",
-        r#"
-            $ tree-sitter-grep -q '(struct_declaration) @c' --language objective-c
-            example.h:4:@property (nonatomic, strong, nullable) NSString *baseURL;
-        "#,
-    );
-}
-
-#[test]
-fn test_objective_c_auto_language() {
-    assert_sorted_output(
-        "objective_c_project",
-        r#"
-            $ tree-sitter-grep -q '(struct_declaration) @c'
-            example.h:4:@property (nonatomic, strong, nullable) NSString *baseURL;
-        "#,
-    );
-}
-
-#[test]
-fn test_objective_c_auto_language_ambiguous_query() {
-    assert_failure_output(
-        "objective_c_project",
-        r#"
-            $ tree-sitter-grep -q '(identifier) @c'
-            File "./example.h" has ambiguous file-type, could be ObjectiveC, C or Cpp. Try passing the --language flag
-        "#,
-    );
-}
-
-#[test]
-fn test_toml() {
+fn test_only_matching() {
     assert_sorted_output(
         "rust_project",
         r#"
-            $ tree-sitter-grep -q '(string) @c' --language toml
-            Cargo.toml:2:name = "rust_project"
-            Cargo.toml:3:version = "0.1.0"
-            Cargo.toml:4:edition = "2021"
+            $ tree-sitter-grep --query-source '(parameter) @c' --language rust --only-matching
+            src/lib.rs:3:left: usize
+            src/lib.rs:3:right: usize
```

Comment:
Since it looks like the underlying printer (and per `ripgrep`'s docs) prints two separate output lines if you do `--only-matching` and there are multiple matches found on the same line (I guess since otherwise it might be printing non-match text between the two matches on that line?), I was curious how it would handle cases with eg multi-line/overlapping matches, so added a couple tests for that below. What it did seems reasonable/not "broken"

Reply:
Ya unless we're opinionated I think that "mimicking a lot of what `ripgrep` does (by leveraging the same `grep`/`ignore` options that it is)" is a good idea/"good enough"